cmd/flux/cmd/test.go

```diff
@@ -300,20 +339,26 @@ func (t *TestRunner) Gather(roots []string) error {
 			if len(baseAST.Files) > 0 {
 				baseAST.Files[0].Name = file.path
 			}
-			tcnames, asts, err := testcase.Transform(ctx, baseAST, modules)
+			tcidens, asts, err := testcase.Transform(ctx, baseAST, modules)
 			if err != nil {
 				return err
 			}
+			pkg := strings.TrimSuffix(baseAST.Package, "_test")
 			for i, astf := range asts {
 				tags, err := readTags(astf)
 				if err != nil {
 					return err
 				}
 				if invalid := invalidTags(tags, mods.Tags(file.module)); len(invalid) != 0 {
-					return errors.Newf(codes.Invalid, "testcase %q, contains invalid tags %v, valid tags are: %v", tcnames[i], invalid, mods.Tags(file.module))
+					return errors.Newf(codes.Invalid, "testcase %q, contains invalid tags %v, valid tags are: %v", tcidens[i].Name, invalid, mods.Tags(file.module))
+				}
+				pkgTest := pkg + "." + tcidens[i].Name
+				if _, ok := seen[pkgTest]; ok {
+					return errors.Newf(codes.AlreadyExists, "duplicate testcase name %q, found in package %q, at locations %v and %v", tcidens[i].Name, pkg, seen[pkgTest].loc.String(), tcidens[i].Loc.String())
```

Comment:
Can we have a test for this? Good use of the `AlreadyExists` code :)

Reply:
Added a test for duplicate testname case